gobblin-metrics/src/main/java/gobblin/metrics/kafka/KafkaReporter.java

```diff
@@ -28,22 +27,31 @@
 import com.google.common.collect.Lists;
 import com.typesafe.config.Config;
 
+import lombok.extern.slf4j.Slf4j;
+
 
 /**
  * Kafka reporter for metrics.
  *
  * @author ibuenros
  */
+@Slf4j
 public class KafkaReporter extends MetricReportReporter {
 
+  public static final String SCHEMA_VERSION_WRITER_TYPE = "metrics.kafka.schemaVersionWriterType";
+
   protected final AvroSerializer<MetricReport> serializer;
   protected final KafkaPusher kafkaPusher;
 
+
   protected KafkaReporter(Builder<?> builder, Config config) throws IOException {
     super(builder, config);
 
-    this.serializer = this.closer.register(
-        createSerializer(new FixedSchemaVersionWriter()));
+    SchemaVersionWriter versionWriter = config.hasPath(SCHEMA_VERSION_WRITER_TYPE) ?
+        SchemaVersionWriter.Type.valueOf(config.getString(SCHEMA_VERSION_WRITER_TYPE)).getVersionWriter() :
```

Comment:
See above comment about using class names or class aliases. The current approach is non-extensible.